redbeams/src/main/java/com/sequenceiq/redbeams/flow/redbeams/start/handler/StartDatabaseServerHandler.java

```diff
@@ -0,0 +1,59 @@
+package com.sequenceiq.redbeams.flow.redbeams.start.handler;
+
+import com.sequenceiq.cloudbreak.cloud.CloudConnector;
+import com.sequenceiq.cloudbreak.cloud.context.AuthenticatedContext;
+import com.sequenceiq.cloudbreak.cloud.context.CloudContext;
+import com.sequenceiq.cloudbreak.cloud.init.CloudPlatformConnectors;
+import com.sequenceiq.flow.event.EventSelectorUtil;
+import com.sequenceiq.flow.reactor.api.handler.EventHandler;
+import com.sequenceiq.redbeams.flow.redbeams.common.RedbeamsEvent;
+import com.sequenceiq.redbeams.flow.redbeams.start.event.StartDatabaseServerFailed;
+import com.sequenceiq.redbeams.flow.redbeams.start.event.StartDatabaseServerRequest;
+import com.sequenceiq.redbeams.flow.redbeams.start.event.StartDatabaseServerSuccess;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import reactor.bus.Event;
+import reactor.bus.EventBus;
+
+@Component
+public class StartDatabaseServerHandler implements EventHandler<StartDatabaseServerRequest> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StartDatabaseServerHandler.class);
+
+    @Inject
+    private CloudPlatformConnectors cloudPlatformConnectors;
+
+    @Inject
+    private EventBus eventBus;
+
+    @Override
+    public String selector() {
+        return EventSelectorUtil.selector(StartDatabaseServerRequest.class);
+    }
+
+    @Override
+    public void accept(Event<StartDatabaseServerRequest> event) {
+        LOGGER.debug("Received event: {}", event);
+        StartDatabaseServerRequest request = event.getData();
+        CloudContext cloudContext = request.getCloudContext();
+        try {
+            CloudConnector<Object> connector = cloudPlatformConnectors.get(cloudContext.getPlatformVariant());
+            AuthenticatedContext ac = connector.authentication().authenticate(cloudContext, request.getCloudCredential());
+
+            connector.resources().startDatabaseServer(ac, request.getDbInstanceIdentifier());
```

Comment:
I think we should check first that do we really need to start the initiate start action or not.
When one of the CB node dies and trying to restart the flow we dont want to initiate the start event twice

Reply:
Its already done in
https://github.com/hortonworks/cloudbreak/blob/fbe8818ef58b368634efdc3e7e16b282435014b9/redbeams/src/main/java/com/sequenceiq/redbeams/service/stack/RedbeamsStartService.java#L43
Should I implement it on AWS level as well?

Reply:
not on aws level but in the abstract flow step